packages/react-select/src/Select.tsx

```diff
@@ -187,7 +187,7 @@ export interface Props<
   /** Whether to enable search functionality */
   isSearchable: boolean;
   /** Async: Text to display when loading options */
-  loadingMessage: (obj: { inputValue: string }) => string | null;
+  loadingMessage: (obj: { inputValue: string }) => ReactNode;
```

Comment:
My initial concern was accessibility but I don't believe that this is something being implemented at present, so this is likely a non-issue.